pkg/appdefinition/appdefinition_test.go

```diff
@@ -3058,3 +3058,52 @@ func TestNestedScopedLabels(t *testing.T) {
 	}}`))
 	assert.Error(t, err)
 }
+
+func TestUserContext(t *testing.T) {
+	appImage, err := NewAppDefinition([]byte(`containers: foo: {
```

Comment:
Might be worth testing smaller/larger values for the inputs, like single digit values and values larger than what we would expect

Reply:
Also maybe `":100"` or `"100:"`

Reply:
In this vein, it might be nice to have fuzz tests https://go.dev/security/fuzz/

Reply:
I added some test cases and tried my hand on fuzzing (never did fuzz testing before - it kinda looks ugly :D)